process/coordinator/sovereignChainProcess.go

```diff
@@ -63,6 +63,16 @@ func (sctc *sovereignChainTransactionCoordinator) CreateMbsAndProcessCrossShardT
 
 	shouldSkipShard := make(map[uint32]bool)
 
+	headerHash, err := core.CalculateHash(sctc.marshalizer, sctc.hasher, hdr)
+	if err != nil {
+		log.Warn("transactionCoordinator.CreateMbsAndProcessCrossShardTransactionsDstMe CalculateHash error",
+			"error", err)
+
+		// we return the nil error here as to allow the proposer execute as much as it can, even if it ends up in a
+		// totally unlikely situation in which it can not marshall a block.
+		return createMBDestMeExecutionInfo.miniBlocks, createMBDestMeExecutionInfo.numTxAdded, false, nil
```

Comment:
return error here. this is critical error.

Reply:
yes. return error.

Reply:
Ok, returned error